launcher/src/main/java/org/apache/spark/launcher/SparkSubmitOptionParser.java

```diff
@@ -57,6 +57,8 @@
   protected final String REPOSITORIES = "--repositories";
   protected final String STATUS = "--status";
   protected final String TOTAL_EXECUTOR_CORES = "--total-executor-cores";
+  protected final String PRINCIPAL = "--principal";
```

Comment:
nit: should probably be moved below with other "YARN-only options". Also, sorting.